src/pages/top-up/Swap.tsx

```diff
@@ -39,10 +43,19 @@ export function Swap({ header }: Props): ReactElement {
     return <Loading />
   }
 
-  const daiToSwap = balance.dai.minusBaseUnits('1')
+  const optimalSwap = balance.dai.minusBaseUnits('1')
+  const lowAmountSwap = new DaiToken(balance.dai.toBigNumber.dividedToIntegerBy(2))
+
+  let daiToSwap: DaiToken
+
+  if (userInputSwap && parseFloat(userInputSwap)) {
```

Comment:
Technically this would allow negative numbers as well, which should not be the case.

Reply:
It also parses '100abc' so it was not a good choice for sanity check